rust_src/src/keymap.rs

```diff
@@ -19,22 +20,24 @@ use crate::{
     lisp::LispObject,
     lists::{nth, setcdr},
     lists::{LispCons, LispConsCircularChecks, LispConsEndChecks},
+    marker::marker_position,
     multibyte::LispStringRef,
     obarray::intern,
     remacs_sys::{
-        access_keymap, apropos_accum, apropos_accumulate, apropos_predicate, copy_keymap_item,
-        describe_vector, make_save_funcptr_ptr_obj, map_char_table, map_keymap_call,
-        map_keymap_char_table_item, map_keymap_function_t, map_keymap_item, map_obarray,
-        maybe_quit, specbind,
+        access_keymap, apropos_accum, apropos_accumulate, apropos_predicate, call2,
+        copy_keymap_item, describe_vector, list2, make_save_funcptr_ptr_obj, map_char_table,
+        map_keymap_call, map_keymap_char_table_item, map_keymap_function_t, map_keymap_item,
+        map_obarray, maybe_quit, menu_item_eval_property, safe_call1, specbind,
     },
-    remacs_sys::{char_bits, current_global_map as _current_global_map, globals, EmacsInt},
+    remacs_sys::{char_bits, current_global_map as _current_global_map,
+    globals, EmacsInt, PURE_P},
     remacs_sys::{
```

Comment:
Whenever you look to import an `F` function check if it has been re-implemented in Rust and call the Rust native function instead. `Flength` for instance. `Fcons` can usually be replicated by making a tuple and calling `into`.